pkg/controllermanager/controller/cloudprofile/cloudprofile_control.go

```diff
@@ -135,11 +138,11 @@ func (c *defaultControl) ReconcileCloudProfile(obj *gardencorev1alpha1.CloudProf
 			}
 			return nil
 		}
-		message := "Can't delete CloudProfile, because Shoots are still referencing it."
-		if len(associatedShoots) != 0 {
```

Comment:
why did you remove this check?

Reply:
If I see that correctly, the check is redundant because if it wasn't true the method would have returned in the previous if block anyways.

Reply:
had to expand, to see the previous block :)